plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+        Returns:
+            dict: A dictionary containing the payload needed to update the Anycast Gateway configuration.
+        Description:
+            This function constructs an update payload for the Anycast Gateway by combining existing values from
+            `anycast_details_in_ccc` with any new values specified in the `anycast` dictionary. If a parameter exists in
+            `anycast`, it is prioritized over the existing value. If a parameter does not exist in `anycast`, the function
+            retains the current value from `anycast_details_in_ccc`.
+            The resulting payload can be used for updating the Anycast Gateway configuration in the Cisco Catalyst Center.
+        """
+
+        vn_name = anycast_details_in_ccc.get("virtualNetworkName")
+        anycast_payload = {
+            "id": anycast_details_in_ccc.get("id"),
+            "fabricId": anycast_details_in_ccc.get("fabricId"),
+            "virtualNetworkName": vn_name,
+            "ipPoolName": anycast_details_in_ccc.get("ipPoolName"),
+            "vlanName": anycast_details_in_ccc.get("vlanName"),
+            "vlanId": anycast_details_in_ccc.get("vlanId"),
+            "isCriticalPool": anycast_details_in_ccc.get("isCriticalPool"),
+            "poolType": anycast_details_in_ccc.get("poolType"),
+            "isIntraSubnetRoutingEnabled": anycast_details_in_ccc.get("isIntraSubnetRoutingEnabled")
+        }
+        params_in_playbook = ["tcp_mss_adjustment", "traffic_type", "security_group_name", "layer2_flooding_enabled",
+                              "fabric_enabled_wireless", "ip_directed_broadcast", "multiple_ip_to_mac_addresses",
+                              "supplicant_based_extended_node_onboarding", "group_policy_enforcement_enabled"]
+
+        anycast_mapping = self.get_anycast_gateway_mapping(vn_name)
+
+        if vn_name == "INFRA_VN":
+            params_to_remove = ["security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless",
+                                "ip_directed_broadcast", "multiple_ip_to_mac_addresses"]
+            for param in params_to_remove:
+                params_in_playbook.remove(param)
+            anycast_payload.pop("isCriticalPool", None)
+            anycast_payload.pop("isIntraSubnetRoutingEnabled", None)
+        else:
+            params_in_playbook.remove("supplicant_based_extended_node_onboarding")
+            params_in_playbook.remove("group_policy_enforcement_enabled")
+            anycast_payload.pop("poolType", None)
+
+        if anycast.get("traffic_type") and anycast_details_in_ccc.get("isCriticalPool") is True:
+            params_in_playbook.remove("traffic_type")
+            anycast_payload["trafficType"] = anycast_details_in_ccc.get("trafficType")
+
+        for param in params_in_playbook:
+            key = anycast_mapping.get(param)
+            if anycast.get(param) is not None:
+                anycast_payload[key] = anycast.get(param)
```

Comment:
self.log("Setting '{0}' to '{1}' in the payload.".format(key, anycast.get(param)), "DEBUG")